src/tactic/core.lean

```diff
@@ -64,14 +66,32 @@ end expr
 namespace interaction_monad
 open result
 
+variables {σ : Type} {α : Type u}
+
+/-- `get_state` returns the underlying state inside an interaction monad, from within that monad. -/
+meta def get_state : interaction_monad σ σ :=
+λ state, success state state
+
+/-- `set_state` sets the underlying state inside an interaction monad, from within that monad. -/
+meta def set_state (state : σ) : interaction_monad σ unit :=
+λ _, success () state
+
+/-- `under_state` applies the given state to `tac`, and returns the result. If `tac` fails, then
```

Comment:
Surely "applies `tac` to the given state"?

Reply:
I was just copying the docstring of `get_result`, which already existed. I'll default to changing them all then.